packages/jest-environment/src/index.ts

```diff
@@ -172,6 +172,12 @@ export interface Jest {
    * local module state doesn't conflict between tests.
    */
   isolateModules(fn: () => void): Jest;
+  /**
+   * `jest.isolateModulesAsync()` is the equivalent of `jest.isolateModules()`, but for
+   * async functions to be wrapped. The caller is expected to `await` the completion of
+   * `isolateModulesAsync`.
+   */
+  isolateModulesAsync(fn: () => void): Promise<void>;
```

Comment:
Should be added to docs as well.

Reply:
Where? I could not find the package for that.

Reply:
See `docs` folder in the root. [CONTRIBUTING.md](https://github.com/facebook/jest/blob/main/CONTRIBUTING.md#additional-workflow-for-any-changes-made-to-website-or-docs) has few lines on that too.

Reply:
Done in 2cf0d73188453435e9776bf1de483713e097b15a